server/webhook.go

```diff
@@ -291,12 +270,100 @@ func (p *parsedJIRAWebhook) fromChangeLog(issue string) (string, string) {
 	return "", ""
 }
 
-func (p *Plugin) notify(ji Instance, parsed *parsedJIRAWebhook, text string) {
+// postEvent posts the event to the channel that subscribed to it
+func (p *Plugin) postEvent(r *http.Request, cfg config, parsed *parsedJIRAWebhook) (int, error) {
+	teamName := r.FormValue("team")
+	if teamName == "" {
+		return http.StatusBadRequest,
+			fmt.Errorf("Request URL: team is empty")
+	}
+
+	channelId := r.FormValue("channel")
+	if channelId == "" {
+		return http.StatusBadRequest,
+			fmt.Errorf("Request URL: channel is empty")
+	}
+
+	user, appErr := p.API.GetUserByUsername(cfg.UserName)
+	if appErr != nil {
+		return appErr.StatusCode, fmt.Errorf(appErr.Message)
+	}
+
+	channel, appErr := p.API.GetChannelByNameForTeamName(teamName, channelId, false)
+	if appErr != nil {
+		return appErr.StatusCode, fmt.Errorf(appErr.Message)
+	}
+
+	initPost := AsSlackAttachment(parsed)
+
+	post := &model.Post{
+		ChannelId: channel.Id,
+		UserId:    user.Id,
+		Props: map[string]interface{}{
+			"from_webhook":  "true",
+			"use_user_icon": "true",
+		},
+	}
+	initPost(post)
+
+	_, appErr = p.API.CreatePost(post)
+	if appErr != nil {
+		return appErr.StatusCode, fmt.Errorf(appErr.Message)
+	}
+
+	return http.StatusOK, nil
+}
+
+// handleNotifications notifies users involved in the event, if they've enabled notifications
+func (p *Plugin) handleNotifications(parsed *parsedJIRAWebhook) error {
+	// This bothers me, to do this for every webhook event...
+	ji, err := p.LoadCurrentJIRAInstance()
+	if err != nil {
+		// It won't break anything if we can't find the Jira Instance here -- we just can't notify anyone.
```

Comment:
I think that if finding the jiraInstance was crucial to the settings command, or some other action by the user, returning an error would definitely be the right thing to do. But here it's in response to a webhook, and the chance of us not finding a current instance would be pretty low. If we didn't find one, it would mean a webhook was left connected and the instance uninstalled -- do we want a million errors being triggered? Probably an error would have thrown or the process stopped way earlier than here...

And the other reason I'm kind of against throwing an error is that I don't know how to mock p.LoadCurrentJIRAInstance in the tests. :) If we want to throw an error, could you show me how?

Reply:
Chatted offline. Going to leave as is for now with a separate ticket to investigate improving the testing situation here.